lib/internal/modules/esm/resolve.js

```diff
@@ -90,6 +90,35 @@ function emitFolderMapDeprecation(match, pjsonUrl, isExports, base) {
   );
 }
 
+function emitLegacyIndexDeprecation(url, packageJSONUrl, base, main) {
+  if (!pendingDeprecation)
+    return;
+  const { format } = defaultGetFormat(url);
+  if (format !== 'module')
+    return;
+  const path = fileURLToPath(url);
+  const pkgPath = fileURLToPath(new URL('.', packageJSONUrl));
+  const basePath = fileURLToPath(base);
+  if (main)
+    process.emitWarning(
+      `Package ${pkgPath} has a "main" field set to ${JSONStringify(main)}, ` +
+      `excluding the full extension to the resolved file at "${
```

Comment:
```suggestion
      `excluding the full filename (including extension) to the resolved file at "${
```
since it might be excluding “index” as well, or point to a nested “main”?

meaning, i have lots of main entries with explicit “./“ or “./index” or “./foo/“ in them, and it’s not clear to me this is covered by the wording